src/components/charts/tooltip-chart/tooltip-chart.js

```diff
@@ -38,20 +38,55 @@ class TooltipChart extends PureComponent {
     return payload.sort(compare);
   };
 
-  renderValue = (y, suffix) => {
-    if (y.payload && y.payload[y.dataKey] !== undefined) {
-      if (Array.isArray(y.payload[y.dataKey])) {
+  renderValue = (y, labelName, suffix) => {
+    if (y.payload && (y.payload[labelName] !== undefined || y.payload.value)) {
+      if (Array.isArray(y.payload[labelName])) {
         return `${this.formatValue(
-          y.payload[y.dataKey][0]
-        )} - ${this.formatValue(y.payload[y.dataKey][1])}${suffix || ''}`;
+          y.payload[labelName][0]
+        )} - ${this.formatValue(y.payload[labelName][1])}${suffix || ''}`;
       }
-      return `${this.formatValue(y.payload[y.dataKey])}${suffix || ''}`;
+      return `${this.formatValue(
+        y.payload[labelName] || y.payload.value
+      )}${suffix || ''}`;
     }
     return 'n/a';
   };
 
-  render() {
+  renderTotal(suffix) {
     const { config, content, showTotal } = this.props;
+    return showTotal && (
+    <div className={cx(styles.label, styles.labelTotal)}>
+      <p>
+            TOTAL
+      </p>
+      <p>
+        {this.getTotal(config.columns.y, content.payload[0], suffix)}
+      </p>
+    </div>
+      );
+  }
+
+  renderIconOrDot(labelName) {
+    const { config } = this.props;
+    const iconColorProp = column => {
+      const { fill, stroke } = config.theme[column];
+      return { style: { fill: fill || stroke, stroke } };
+    };
+    const labelConfig = config.theme[labelName];
+    return labelConfig.icon ? (
+      <span className={styles.labelIcon}>
+        <Icon icon={labelConfig.icon} {...iconColorProp(labelName)} />
+      </span>
+) : (
+  <span
+    className={styles.labelDot}
+    style={{ backgroundColor: labelConfig && labelConfig.stroke }}
+  />
+);
+  }
+
+  render() {
```

Comment:
Would make sense to add `TooltipChart` component to StyleGuidist? Just to have the component documented and know what's the `config` shape that we have to pass to the chart to have the tooltip properly working.